corokafka/impl/corokafka_consumer_manager_impl.cpp

```diff
@@ -961,33 +974,46 @@ ConsumerManagerImpl::deserializeMessage(ConsumerTopicEntry& entry,
     int num = 0;
     const cppkafka::HeaderList<cppkafka::Header<cppkafka::Buffer>>& kafkaHeaders = kafkaMessage.get_header_list();
     for (const auto& header : kafkaHeaders) {
-        try {
-            const TypeErasedDeserializer::HeaderEntry& headerEntry = deserializer._headerDeserializers.at(header.get_name());
-            headers[headerEntry._pos].first = header.get_name();
-            headers[headerEntry._pos].second = cppkafka::CallbackInvoker<Deserializer>("header_deserializer",
-                                                                                       *headerEntry._deserializer,
-                                                                                       entry._consumer.get())
-                                                                         (toppar, header.get_value());
-            if (headers[headerEntry._pos].second.empty()) {
-                // Decoding failed
-                de._error = RD_KAFKA_RESP_ERR__VALUE_DESERIALIZATION;
-                de.setError(DeserializerError::Source::Header);
-                de._headerNum = num;
-                std::ostringstream oss;
-                oss << "Failed to deserialize header: " << header.get_name();
-                report(entry, cppkafka::LogLevel::LogErr, RD_KAFKA_RESP_ERR__VALUE_DESERIALIZATION, oss.str(), &kafkaMessage);
-                break;
-            }
-        }
-        catch (const std::exception& ex) {
+        auto deserIter = deserializer._headerDeserializers.find(header.get_name());
+        if (deserIter == deserializer._headerDeserializers.end()) {
+            std::ostringstream oss;
+            oss << "Found unknown header [" << header.get_name() << "] in topic " << toppar.get_topic();
             if (entry._skipUnknownHeaders) {
-                report(entry, cppkafka::LogLevel::LogWarning, {}, ex.what(), &kafkaMessage);
+                //Check if we have already warned about this unknown header
```

Comment:
I'm rather ambivalent about this part of the change; it's nice to not have the spam/spew in the logs, but that also decreases the odds that the library client will notice this error is coming up.
Unlike in some (entirely internal) libraries where we control the logging directly, we don't have a known place where the logs should be, which makes it hard for us to set up a ticket for the client.

Maybe instead of logging each unknown header once, we try to log all unknown headers for a given message once & together? i.e. "Found unknown header(s) [a, b] in topic TOPIC".

Basically my concern is that when skip unknown headers is on (which it is by default), we're doing work to reduce spam in the log, but we may be reducing it so much that clients may not even notice the skipped header, which may mask bugs in their configuration (there's no difference between an entirely unconfigured header, and a mis-named header).

That's my two cents or so, but I'm comfortable being outvoted here by other maintainers.